README.md

```diff
@@ -179,6 +179,38 @@ make timed_run
 make stop_slaves  # mac: killall python
 ```
 
+#### Profiling
+
+In case you need to profile different the application to know which part of the code are more time consuming, there are two different profilers available. Both will store the profiling output in a directory that's determined by the `PROFILE_DIR` env variable. If  this variable is not set, the output files will be store in a folder called profiler inside the OS temp folder (`/tmp/profiler` usually)
+
+##### All requests profiling mode
+
+This mode will create interactive html files with all the profiling information for every request the Studio server receives. The name of the files will contain the total execution time, the endpoint  name and a timestamp.
+
+To activate it an env variable called `PROFILE_STUDIO_FULL` must be set.
+
+Example of use:
+
+`PROFILE_STUDIO_FULL=y yarn runserver`
+
+##### Endpoint profiling mode
+
+When using the all requests mode it's usual that the profile folder is soon full of information for requests that are not interesting for the developer and it's hard to find the files for some specific endpoints.
+
+If an env variable called `PROFILE_STUDIO_FILTER` is used, the profiler will be executed only on the http requests containing the text stated by the variable.
+
+Example of use:
+
+`PROFILE_STUDIO_FILTER=edit yarn localprodserver`
+
+For this case, only html requests having the word edit in his request path will be profiled. The profile folder will not have html files, but binary dump files (with the timestamp as filename) of the profiler information that can be later seen by different profiling tools (`snakeviz` that can be installed using pip is recommended). Also while the server is running,  the ten most time consuming lines of code of the filtered request will be shown in the console where Studio has been launched.
+
+Example of snakeviz use:
+
+`snakeviz /tmp/profile/studio\:20200909161405011678.prof`
```

Comment:
I only am seeing *.html files generated in my `/tmp/profile` folder.

`snakeviz /tmp/profile` opens the server and a directory list, but when I open any of the HTML files within snakeviz, I get:

```
Traceback (most recent call last):
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/snakeviz/main.py", line 32, in get
    s = Stats(profile_name)
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/pstats.py", line 96, in __init__
    self.init(arg)
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/pstats.py", line 110, in init
    self.load_stats(arg)
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/pstats.py", line 124, in load_stats
    self.stats = marshal.load(f)
ValueError: bad marshal data (unknown type code)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/tornado/web.py", line 1590, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/home/jacob/.pyenv/versions/3.7.7/lib/python3.7/site-packages/snakeviz/main.py", line 34, in get
    raise RuntimeError('Could not read %s.' % profile_name)
RuntimeError: Could not read /tmp/profile/0.118s _api_contentnode?id__in=3632dbc3e54d44fcb24cd439a7cb7696 1599680379.html.
```

Reply:
@nucleogenesis , as commented above, if you're using _All requests profiling mode_  mode, snakeviz won't be needed as the output are html files. snakeviz is needing if you set the env variables for the _Endpoint profiling mode _
It's either one mode or the other, both can't be active and if you set both variables, _All requests profiling mode_ will be used.